### PR TITLE
Resolves #45 - implementation of GPU cache

### LINKED ISSUE #45: Implement GPU cache for Octree
Store Voxelements, nodes and children in a fixed size cache array. The entries in the cache shall be stored inside the GPU octree structs, and shall be invalidated upon overwriting.

The result is an octree with a fixed, well defined size, where data is loaded on request. 
This makes it possible to:
- store an array of octrees inside the GPU - possible way to store a panorama
- to properly divide the data update of the tree into relatively smaller segments(therefore adequate for loop-by-loop update)
- Implement lazy-loading to the GPU, therefore this issue is a parallell to #3
- Support different LOD-s through nodes: higher level nodes might have voxel data ( see #2 )

### PR DESCRIPTION
**Pull request by davids91**

### CODE REVIEW

File: examples/minecraft.rs [hunk 1]
```diff
@@ -152,9 +148,14 @@ fn rotate_camera(
 #[cfg(feature = "bevy_wgpu")]
 fn handle_zoom(
     keys: Res<ButtonInput<KeyCode>>,
-    mut viewing_glass: ResMut<ShocoVoxViewingGlass>,
     mut angles_query: Query<&mut DomePosition>,
+    svx_data: Option<ResMut<ShocoVoxRenderData>>,
 ) {
+    if let Some(ref svx_data) = svx_data {
+        if svx_data.is_changed() {
+            println!("changed!! --> {:?}", svx_data.node_children[0..2].to_vec());
```
Comment (nerdachse) [inline examples/minecraft.rs:156]: At some point we may want to add `tracing` or the like to our dependencies

**Review by nerdachse — APPROVED** [review]

File: src/octree/raytracing/bevy/types.rs [hunk 2]
```diff
@@ -82,6 +85,7 @@ pub struct ShocoVoxRenderData {
     /// | Byte 3  | unused                                                    |
     /// `=====================================================================`
     /// *(1) Only first bit is used in case leaf is uniform
```
Comment (nerdachse) [inline src/octree/raytracing/bevy/types.rs:87]: Am I blind and just not seeing this footnote being referenced anywhere? oO

Comment (davids91) [reply src/octree/raytracing/bevy/types.rs:87]: it's not in the changes :)

File: examples/minecraft.rs (hunk 1 above)
Comment (davids91) [reply examples/minecraft.rs]: That's just some temporary debug display, but I agree on the tracing dependencies!

**Pull request merged**